shared/src/commonMain/kotlin/com/devil/phoenixproject/presentation/components/DeleteProfileDialog.kt

```
@@ -25,9 +33,21 @@ fun DeleteProfileDialog(profile: UserProfile, profileRepository: UserProfileRepo
             TextButton(
                 onClick = {
                     scope.launch {
-                        profileRepository.deleteProfile(profile.id)
+                        // Issue #393: Unhandled exceptions in profile deletion caused
+                        // SIGABRT on iOS (Kotlin/Native abort() on uncaught exception).
+                        try {
+                            profileRepository.deleteProfile(profile.id)
+                            onDismiss()
+                        } catch (e: kotlin.coroutines.cancellation.CancellationException) {
+                            throw e
+                        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)